openquake/commonlib/readinput.py

```diff
@@ -833,11 +834,33 @@ def get_pmap_from_csv(oqparam, fnames):
     return mesh, ProbabilityMap.from_array(data, range(len(mesh)))
 
 
-# used in utils/reduce_sm and utils/extract_source
+tag2code = {'ar': b'A',
+            'mu': b'M',
+            'po': b'P',
+            'si': b'S',
+            'co': b'C',
+            'ch': b'X',
+            'no': b'N'}
+
+
+# used in oq reduce_sm and utils/extract_source
 def reduce_source_model(smlt_file, source_ids, remove=True):
     """
-    Extract sources from the composite source model
+    Extract sources from the composite source model.
+
+    :param smlt_file: path to a source model logic tree file
+    :param source_ids: dictionary source_id -> source_code
```

Comment:
`source_ids` can also be a list of source IDs as in extract_source (see line 860)